### PR TITLE
hack,test: Gather the OLM registry resources during e2e runs

### PR DESCRIPTION
**Pull request by timflannagan1**
Adds a hack script that's responsible for gathering the CatalogSource and the corresponding PackageManifest custom resources that were used to deploy a custom version of Metering.

Updates the test helpers, adding a function that calls that bash script is responsible for setting the correct environment variables based on the user-provided e2e flags. The main_test.go e2e driver will run this gather registry artifacts test helper function when the e2e suite has been instantiated without the `--run-dev-setup` flag.

### CODE REVIEW

File: test/testhelpers/helpers.go [hunk 1]
```diff
@@ -289,3 +290,28 @@ func ExecPodCommandWithOptions(config *rest.Config, client kubernetes.Interface,
 
 	return stdoutBuf, stderrBuf, nil
 }
+
+func GatherRegistryResources(outputPath, scriptPath, name, namespace string) error {
+	path, err := exec.LookPath(scriptPath)
```
Comment (timflannagan1) [inline test/testhelpers/helpers.go:295]: Just for reviewers. this `exec.LookPath` is a pretty nice utility function that I didn't realize existed before re-reading that exec package documentation. It essentially acts as a `os.Stat` operation to ensure the path to the binary or script exists, which is nice. We may want to update the rest of our instances of this package to also use this function.

Comment (timflannagan1) [reply test/testhelpers/helpers.go:295]: Also, the link to that function documentation: https://golang.org/pkg/os/exec/#LookPath.

File: hack/gather-registry-artifacts.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#! /bin/bash
+
+set -eou pipefail
+set -x
+
+CATALOG_SOURCE_NAMESPACE=${CATALOG_SOURCE_NAMESPACE:=openshift-marketplace}
+CATALOG_SOURCE_NAME=${CATALOG_SOURCE_NAME:=redhat-operators}
+
+OUTPUT_DIRECTORY=${OUTPUT_DIRECTORY:=$(mktemp -d)/resources}
```
Comment (bentito) [inline hack/gather-registry-artifacts.sh:9]:
```suggestion
OUTPUT_DIRECTORY=${OUTPUT_DIRECTORY:=$(mktemp -ud)/resources}
```
`-u` is dry run, avoids creating a directory twice

File: test/testhelpers/helpers.go (hunk 1 above)
Comment (bentito) [reply test/testhelpers/helpers.go:295]: 👍

File: test/e2e/main_test.go [hunk 3]
```diff
@@ -124,7 +133,26 @@ func testMainWrapper(m *testing.M) int {
 		df.Logger.Fatalf("Failed to create the CatalogSource custom resource using the %s index image: %v", indexImage, err)
 	}
 	if !df.RunDevSetup {
-		defer df.DeleteRegistryResources(registryProvisioned, catalogSourceName, catalogSourceNamespace)
+		var (
+			errArr []string
```
Comment (bentito) [inline test/e2e/main_test.go:137]:
```suggestion
			errors []string
```

Comment (bentito) [reply test/e2e/main_test.go:137]: unless that's a keyword, idk reads easier to me. I can look and see the type so having in the name doesn't help me much.

**Pull request merged**